vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -514,8 +480,80 @@ static void run_dijkstra(int start_node_ind,
     }
 }
 
-void ConnectionBoxMapLookahead::compute(const std::vector<t_segment_inf>& segment_inf, const std::string& lookahead_search_locations) {
-    compute_connection_box_lookahead(segment_inf, lookahead_search_locations);
+void ConnectionBoxMapLookahead::compute(const std::vector<t_segment_inf>& segment_inf,
+                                        const std::string& search_locations) {
+    vtr::ScopedStartFinishTimer timer("Computing connection box lookahead map");
+
+    size_t num_segments = segment_inf.size();
+    find_start_inodes_for_segment_types(num_segments);
+
+    /* free previous delay map and allocate new one */
+    auto& device_ctx = g_vpr_ctx.device();
+    cost_map_.set_counts(segment_inf.size(),
+                         device_ctx.connection_boxes.num_connection_box_types());
+
+    /* run Dijkstra's algorithm for each segment type & channel type combination */
+#if defined(VPR_USE_TBB)
+    tbb::mutex cost_map_mutex;
+    tbb::parallel_for(size_t(0), size_t(num_segments), [&](int iseg) {
```

Comment:
Why not make a function and then call it either via par_for or the typical for loop?

Reply:
I think a better idea is to write `possibly_parallel_for()` to remove the ugly `#if defined(...)`.